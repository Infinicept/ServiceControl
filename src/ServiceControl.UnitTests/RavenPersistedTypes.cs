namespace ServiceControl.UnitTests
{
    using System;
    using System.Linq;
    using NUnit.Framework;
    using Particular.Approvals;
    using Particular.ServiceControl;
    using Raven.Client.Indexes;

    class RavenPersistedTypes
    {
        [Test]
        public void Verify()
        {
            var documentTypes = typeof(Bootstrapper).Assembly.GetTypes()
                .Where(type => typeof(AbstractIndexCreationTask).IsAssignableFrom(type))
                .SelectMany(indexType => indexType.BaseType?.GenericTypeArguments)
                .Distinct();

            var documentTypeNames = string.Join(Environment.NewLine, documentTypes.Select(t => t.AssemblyQualifiedName));

            Approver.Verify(documentTypeNames);
        }
    }
}