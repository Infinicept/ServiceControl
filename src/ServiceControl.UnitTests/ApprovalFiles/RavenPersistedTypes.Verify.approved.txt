ServiceControl.SagaAudit.SagaListIndex+Result, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageFailures.FailedMessage, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.FailureGroupView, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.FailureGroupMessageView, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageFailures.GroupComment, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.FailedMessageRetry, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.RetryBatch, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.RetryBatchGroup, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Monitoring.KnownEndpoint, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageFailures.QueueAddress, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Operations.FailedErrorImport, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Operations.FailedAuditImport, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.EventLog.EventLogItem, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageAuditing.ProcessedMessage, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.CustomChecks.CustomCheck, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.CompositeViews.Messages.MessagesViewIndex+SortAndFilterOptions, ServiceControl, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null